packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java

```diff
@@ -265,11 +265,17 @@ public void onCodeAutoRetrievalTimeOut(String verificationId) {
 
   private void handleLinkWithEmailAndPassword(
       MethodCall call, Result result, FirebaseAuth firebaseAuth) {
+    if (firebaseAuth.getCurrentUser() == null) {
```

Comment:
nit: Can we store this as a local variable so we don't need to fetch it again below? This applies to all of these safeguards.